pkg/platform/update-api/platform.go

```diff
@@ -0,0 +1,163 @@
+package update_api
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/Masterminds/semver"
+	"github.com/bottlerocket-os/bottlerocket-update-operator/pkg/logging"
+	"github.com/bottlerocket-os/bottlerocket-update-operator/pkg/platform"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"time"
+)
+
+// Assert Update-API as a platform implementor.
+var _ platform.Platform = (*Platform)(nil)
+
+type Platform struct {
+	log        logging.Logger
+	httpClient http.Client
+}
+
+func New() (*Platform, error) {
+	return &Platform{log: logging.New("platform"), httpClient: http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", bottlerocketApiSock)
+			},
+		},
+		// Set a 10 second timeout for all requests
+		Timeout: 10 * time.Second,
+	}}, nil
+}
+
+type statusResponse struct {
+	osVersion *semver.Version
+}
+
+func (sr *statusResponse) OK() bool {
+	// Bottlerocket OS version needs to be at least 0.4.1 to support the Update API
+	constraint, _ := semver.NewConstraint(">= 0.4.1")
+	return constraint.Check(sr.osVersion)
+}
+
+// Try to determine if the Bottlerocket version at least 0.4.1
+func (p Platform) Status() (platform.Status, error) {
+	response, err := p.httpClient.Get("http://unix" + "/os")
+	if err != nil {
+		p.log.WithError(err).Error("error when trying to GET '/os'")
+		return nil, err
+	}
+	var osInfo interface{}
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(body, osInfo)
+	if err != nil {
+		return nil, err
+	}
+	// Assert 'version_id' in the OS info json output as string
+	osVersion, err := semver.NewVersion(osInfo.(map[string]interface{})["version_id"].(string))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse 'version_id' field as semver")
+	}
+	return &statusResponse{osVersion: osVersion}, nil
+}
```

Comment:
I think it would be easier if you directly unmarhsalled into a type with that field.  Like this:

```go
type osInfoJSON {
	versionID string `json:"version_id"`
}

//...

var osInfo osInfoJSON
err = json.Unmarshal(body, &osInfo)
// err

osVersion, err := semver.NewVersion(osInfo.versionID)
// err
```